### PR TITLE
Refactor unit tests to use more convenient doctest assertion macros (Part 2)

### PR DESCRIPTION
**Pull request by kkarbowiak**
Also, removed some now redundant code and adjusted the expected error messages. The new messages reflect what would be communicated if the old assertion macros were reordered (first `CHECK_THROWS_WITH` and then `CHECK_THROWS_AS`).

### CODE REVIEW

File: test/src/unit-regression1.cpp [hunk 1]
```diff
@@ -1413,24 +1411,14 @@ TEST_CASE("regression tests 1")
         auto p1 = R"([{"op": "move",
                        "from": "/one/two/three",
                        "path": "/a/b/c"}])"_json;
-        CHECK_THROWS_AS(model.patch(p1), json::out_of_range&);
+        CHECK_THROWS_WITH_AS(model.patch(p1),
+                             "[json.exception.out_of_range.403] key 'a' not found", json::out_of_range&);
 
-        auto p2 = R"([{"op": "move",
```
Comment (nlohmann) [inline test/src/unit-regression1.cpp:1418]: Why did you remove the test for `move` here?

Comment (kkarbowiak) [reply test/src/unit-regression1.cpp:1418]: Not really removed. The test was executed twice: once to assert an exception is thrown, and second time to assert the exception's message. The new assertion macro does both checks at once, so I removed the redundant code.

Comment (gregmarr) [reply test/src/unit-regression1.cpp:1418]: I had the exact same question, and then I realized that my confusion was mostly due to the way that the diff was aligned.  :)

Comment (kkarbowiak) [reply test/src/unit-regression1.cpp:1418]: Yes, the diff aligning is a bit unfortunate in this case. The situation is more clear if you inspect the changes by commits, though.

**Review by nlohmann — APPROVED** [review]
Looks good to me.

**Comment by kkarbowiak** [comment]
Thank you for taking the time to review and accept the changes.
